plugins/pantheon-files-cloud/plugin.vala

```diff
@@ -0,0 +1,127 @@
+/***
+    Copyright (c) 2018 elementary LLC <https://elementary.io>
+
+    Pantheon Files is free software; you can redistribute it and/or
+    modify it under the terms of the GNU General Public License as
+    published by the Free Software Foundation; either version 2 of the
+    License, or (at your option) any later version.
+
+    Pantheon Files is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+    Author(s):  Fernando da Silva Sousa <wild.nando@gmail.com>
+***/
+
+public class Marlin.Plugins.Cloud.Plugin : Marlin.Plugins.Base {
+    Marlin.AbstractSidebar? sidebar;
+    CloudProviders.Collector collector;
+    CloudProviders.Provider[] providers_connected;
```

Comment:
Since this is an array made out of non-simple types, I would advise to use `Gee.ArrayList` or `Gee.LinkedList` here. It's not necessary in current state of this branch though.

Reply:
Changed to ArrayList.